packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -255,7 +285,45 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
 
   // ignore unused private method (it's actually used in Stage)
   // @ts-ignore
-  private _generateOutputArtifactName(stage: cpapi.IStage, action: cpapi.Action): string {
+  private _attachAction(stage: Stage, action: actions.Action): void {
+    // handle cross-region Actions here
+    if (action.region) {
+      // get the region the Pipeline itself is in
+      const pipelineStack = cdk.Stack.find(this);
+      const pipelineRegion = pipelineStack.requireRegion(
+        "You need to specify an explicit region when using CodePipeline's cross-region support");
+      const pipelineAccount = pipelineStack.requireAccountId(
+        "You need to specify an explicit account when using CodePipeline's cross-region support");
+      const app = pipelineStack.parent === undefined ? undefined : pipelineStack.parent as cdk.App;
+
+      // if we don't have an ArtifactStore generated for this non-Pipeline Region yet
+      if (!this.artifactStores[action.region] && action.region !== pipelineRegion) {
+        let replicationBucketName = this.crossRegionReplicationBuckets[action.region];
+        if (!replicationBucketName) {
+          const bootstrapStack = new CrossRegionScaffoldStack(app, {
+            region: action.region,
+            account: pipelineAccount,
+          });
+          this._crossRegionScaffoldStacks[action.region] = bootstrapStack;
+          replicationBucketName = bootstrapStack.replicationBucket;
+        }
+
+        const replicationBucket = s3.BucketRef.import(this, 'CrossRegionCodePipelineReplicationBucket-' + action.region, {
+          bucketName: replicationBucketName,
+        });
+        replicationBucket.grantReadWrite(this.role);
```

Comment:
Would this be de-duplicated if multiple actions target the same region, or are we going to see multiple statements with the same content?

Reply:
No, we would not see multiple statements with the same content, it would be de-duplicated. That's what the `this.artifactStores[action.region]` condition is there for.